### PR TITLE
feat: Add cypress tests for ProviderCard

### LINKED ISSUE #1658: [NV-925] - Add cypress component test to ProviderCard component
## Why?

To help increase our test coverage within the `novu/web` project, we can add the newly released [cypress component tests](https://docs.cypress.io/guides/component-testing/testing-react).

## How?

A new component test file should be added located near the original source file(see example at `apps/web/src/design-system/button/button.cy.tsx` and could be run after running `npm run start` from the root of the monorepo and specifying "Run Tests" and "Open Cypress", after which select the "Component testing" in the cypress screen.

Write the new test and make sure it passes all assertions.

The component is located at: apps/web/src/pages/integrations/components/ProviderCard.tsx` `

### PR DESCRIPTION
**Pull request by shtelzerartem**
- **What kind of change does this PR introduce?** 
Adds cypress tests for `apps/web/src/pages/integrations/components/ProviderCard.tsx` component

- **Why was this change needed?** 
Resolves #1658

### CODE REVIEW

**Review by p-fernandez — APPROVED** [review]
🌟

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    </TestWrapper>
+  );
+
+  cy.getByTestId(providerCardTestId)
+    .find('img')
+    .should('have.attr', 'src', `/static/images/providers/${colorScheme}/${provider.logoFileName[`${colorScheme}`]}`)
+    .should('have.attr', 'alt', provider.displayName);
+});
+
+it('dark theme component', () => {
```
Comment (scopsy) [inline apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:52]: @shtelzerartem when writing tests we usually try to construct the message to be read in the following way:
```suggestion
it('should change the image based on the colorScheme selected by user', () => {
```

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    </TestWrapper>
+  );
+
+  cy.getByTestId(providerCardTestId)
+    .find('img')
+    .should('have.attr', 'src', `/static/images/providers/${colorScheme}/${provider.logoFileName[`${colorScheme}`]}`)
+    .should('have.attr', 'alt', provider.displayName);
+});
+
+it('connected component', () => {
```
Comment (scopsy) [inline apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:69]:
```suggestion
it('should show a connected state when the provider is configured', () => {
```

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  cy.get('@clickSpy').its('callCount').should('equal', 1);
+  cy.get('@clickSpy').should('have.been.calledWith', true, false, connectedProvider);
+
+  cy.getByTestId(providerCardTestId).should('not.contain', 'COMING SOON').should('not.contain', 'BETA');
+  cy.getByTestId(providerCardTestId).get('button').should('not.be.exist');
+  cy.getByTestId('provider-card-settings-svg').should('exist');
+  cy.getByTestId('card-status-bar-active').should('exist');
+});
+
+it('not connected component', () => {
```
Comment (scopsy) [inline apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:89]:
```suggestion
it('should show the default state when not connected', () => {
```

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  cy.get('@clickSpy').its('callCount').should('equal', 1);
+  cy.get('@clickSpy').should('have.been.calledWith', true, true, provider);
+
+  cy.getByTestId(providerCardTestId).should('not.contain', 'COMING SOON').should('not.contain', 'BETA');
+  cy.getByTestId(providerCardTestId).get('button').should('exist').should('not.have.attr', 'disabled');
+  cy.getByTestId('provider-card-settings-svg').should('not.be.exist');
+  cy.getByTestId('card-status-bar-active').should('not.be.exist');
+});
+
+it('coming soon component', () => {
```
Comment (scopsy) [inline apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:108]:
```suggestion
it('should display a coming soon banner when applied', () => {
```

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+  cy.getByTestId(providerCardTestId).click();
+  cy.get('@clickSpy').should('not.have.been.calledWith');
+
+  cy.getByTestId(providerCardTestId).should('contain', 'COMING SOON').should('not.contain', 'BETA');
+  cy.getByTestId(providerCardTestId).get('button').should('exist').should('have.attr', 'disabled');
+  cy.getByTestId('card-status-bar-active').should('not.be.exist');
+});
+
+it('beta version connected component', () => {
```
Comment (scopsy) [inline apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:129]:
```suggestion
it('should display the beta version banner when in beta', () => {
```

**Review by scopsy — COMMENTED** [review]
Thank you for the contribution @shtelzerartem, added a suggestion in terms of naming the actual tests cases to conform with our practices :)

**Comment by shtelzerartem** [comment]
@scopsy Thanks! I will change it to your suggestions!

File: apps/web/src/pages/integrations/components/ProviderCard.cy.tsx (hunk 1 above)
Comment (shtelzerartem) [reply apps/web/src/pages/integrations/components/ProviderCard.cy.tsx:52]: I will change it to this, ok? 
For dark:
```javascript
it('should change the image based on the colorScheme (dark) selected by user', () => {
```
For light:
```javascript
it('should change the image based on the colorScheme (light) selected by user', () => {
```

**Review by scopsy — APPROVED** [review]
Thank you so much for your contribution! Really appreciate it 🙏

**Review by p-fernandez — APPROVED** [review]
🌟

**Pull request merged**